sdk/core/core-client-rest/src/multipart.ts

```diff
@@ -0,0 +1,186 @@
+   */
+  dispositionType?: string;
+
+  /**
+   * The field name associated with this part. This value will be appended to the Content-Disposition header if dispositionType is set.
+   */
+  name?: string;
+
+  /**
+   * The file name of the content if it is a file. This will be appended to the Content-Disposition header value if dispositionType is set.
+   */
+  filename?: string;
+
+  /**
+   * The multipart headers for this part of the multipart body. Values of the Content-Type and Content-Disposition headers set in the headers bag
+   * will take precedence over those computed from the request body or the contentType, dispositionType, name, and filename fields on this object.
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * The body of this part of the multipart request.
+   */
+  body?: unknown;
+}
+
+type MultipartBodyType = BodyPart["body"];
+
+/**
+ * Get value of a header in the part descriptor ignoring case
+ */
+function getHeaderValue(descriptor: PartDescriptor, headerName: string): string | undefined {
+  if (descriptor.headers) {
+    const actualHeaderName = Object.keys(descriptor.headers).find(
+      (x) => x.toLowerCase() === headerName.toLowerCase(),
+    );
+    if (actualHeaderName) {
+      return descriptor.headers[actualHeaderName];
+    }
+  }
+
+  return undefined;
+}
+
+function getPartContentType(descriptor: PartDescriptor): string | undefined {
+  const contentTypeHeader = getHeaderValue(descriptor, "content-type");
+  if (contentTypeHeader) {
+    return contentTypeHeader;
+  }
+
+  // Special value of null means content type is to be omitted
+  if (descriptor.contentType === null) {
+    return undefined;
+  }
+
+  if (descriptor.contentType) {
+    return descriptor.contentType;
+  }
+
+  const { body } = descriptor;
+
+  if (body === null || body === undefined) {
+    return undefined;
+  }
+
+  if (typeof body === "string" || typeof body === "number" || typeof body === "boolean") {
+    return "text/plain; charset=UTF-8";
+  }
+
+  if (body instanceof Blob) {
+    return body.type || "application/octet-stream";
+  }
+
+  if (isBinaryBody(body)) {
+    return "application/octet-stream";
+  }
+
+  // arbitrary non-text object -> generic JSON content type by default. We will try to JSON.stringify the body.
+  return "application/json; charset=UTF-8";
+}
+
+function getContentDisposition(descriptor: PartDescriptor): string | undefined {
+  const contentDispositionHeader = getHeaderValue(descriptor, "content-disposition");
+  if (contentDispositionHeader) {
+    return contentDispositionHeader;
+  }
+
+  if (
+    descriptor.dispositionType === undefined &&
+    descriptor.name === undefined &&
+    descriptor.filename === undefined
+  ) {
+    return undefined;
+  }
+
+  const dispositionType = descriptor.dispositionType ?? "form-data";
+
+  let disposition = dispositionType;
+  if (descriptor.name) {
+    disposition += `; name="${descriptor.name}"`;
+  }
+
+  const filename =
```

Comment:
would we trust the name set in Blob also?

Reply:
`Blob` doesn't have a name property.

We don't set the default value of the filename to "blob" anymore with this change; the filename is left unspecified if it's not provided. This brings us in line with the other languages -- previously, we were unable to do that due to limitations of the browser implementation of `FormData` and `fetch`. Since we're using our own multipart implementation we no longer have have this limitation.

Reply:
We could set the default to `blob` just for the case of `File` and `Blob` instances without a file name though (and not specify a filename for other binary types when a file name is not provided)? I'm not sure I am a fan of that, it feels inconsistent.

Reply:
I think we could confirm this with other languages and it should be an consistent behavior cross languages.